keanu-python/tests/test_sampling.py

```diff
@@ -58,7 +62,17 @@ def test_down_sample_interval(net: BayesNet) -> None:
         net=net, sample_from=net.get_latent_vertices(), draws=draws, down_sample_interval=down_sample_interval)
 
     expected_num_samples = draws / down_sample_interval
-    assert all(len(vertex_samples) == expected_num_samples for vertex_id, vertex_samples in samples.items())
+    assert all(len(vertex_samples) == expected_num_samples for label, vertex_samples in samples.items())
+
+
+@pytest.mark.mpl_image_compare(filename='test_sampling_windows.png' if os.name == 'nt' else 'test_sampling_linux.png')
```

Comment:
This SO post suggests `platform.system() == 'Windows'` is a better catch-all check... not a big deal though, since all we really care about is that the test passes on CI.

https://stackoverflow.com/questions/1854/python-what-os-am-i-running-on

Reply:
Another optional suggestion: if we have several of these tests, it would be good to encapsulate the logic in a method `def construct_os_dependent_file_name(filename: str) -> str`

Reply:
Spoken to some people offline, we decided to not to image tests altogether for now. The reasoning was it's sufficient to test against axes. If we ever come to requiring e2e tests for this in the future, we can revisit this test method later.